File: core/src/main/dc/targetman/mechanics/StickActions.java
package dc.targetman.mechanics;

import dc.targetman.epf.parts.MovementPart;
import dc.targetman.epf.parts.WeaponPart;
import dclib.epf.Entity;

public final class StickActions {

	private StickActions() {
	}

	public static final void move(final Entity entity, final Direction direction) {
		entity.get(MovementPart.class).setDirection(direction);
	}

	public static final void jump(final Entity entity) {
		entity.get(MovementPart.class).setTryJumping(true);
	}

	public static final void aim(final Entity entity, final int direction) {
        // TODO:
//		entity.get(WeaponPart.class).setAimDirection(direction);
    }

	public static final void trigger(final Entity entity) {
		entity.get(WeaponPart.class).setTriggered(true);
	}

}

File: core/src/main/dc/targetman/mechanics/StickActions.kt
package dc.targetman.mechanics

import dc.targetman.epf.parts.MovementPart
import dc.targetman.epf.parts.WeaponPart
import dclib.epf.Entity

object StickActions {
    fun move(entity: Entity, direction: Direction) {
        entity[MovementPart::class].direction = direction
    }

    fun jump(entity: Entity) {
        entity[MovementPart::class].tryJumping = true
    }

    fun aim(entity: Entity, direction: Int) {
        // TODO:
        //		entity.get(WeaponPart.class).setAimDirection(direction);
    }

    fun trigger(entity: Entity) {
        entity[WeaponPart::class].setTriggered(true)
    }
}